uaclient/entitlements/repo.py

```diff
@@ -165,6 +165,26 @@ def application_status(self) -> "Tuple[ApplicationStatus, str]":
             "{} is not configured".format(self.title),
         )
 
+    def _check_apt_url_delta_should_be_applied(self, delta_apt_url):
```

Comment:
We may want this to be more generic than across apt deltas right? I was thinking this would be built in as a check_apt_url_is_applied and not really just dealing with deltas. Then we can check current contract state for our running ubuntu release, and validate that the aptURL specified is already applied to the self.repo_list_file_tmpl.format(name=self.name). If not applied, then apply it.

Reply:
Yes, we could make the name of the function more generic. The reason I was so specific here is just because this method is only being used on the `process_contract_deltas` method. But I do agree we should aim at being more generic here.